lib/auth/authentication_backend.rb

```diff
@@ -31,6 +33,28 @@ def authentication_flow(params)
     raise AuthenticationBackendException.new("No Response Back from Authentication Service to process")
   end
 
+  def request_password_reset(params)
+    client.forgot_password(
+      client_id: cognito_client_id,
+      username: params[:email]
+    )
+  rescue Aws::CognitoIdentityProvider::Errors::NotAuthorizedException => e
+    raise UserBadStateError.new(e)
+  rescue Aws::CognitoIdentityProvider::Errors::UserNotFoundException => e
+    raise UserGroupNotFoundException.new(e)
+  rescue Aws::CognitoIdentityProvider::Errors::LimitExceededException => e
+    raise ToManyAttemptsError.new(e)
+  end
+
+  def reset_password(params)
+    client.confirm_forgot_password(
+      client_id: cognito_client_id,
+      username: params[:email],
+      confirmation_code: params[:code],
+      password: params[:password]
+      )
```

Comment:
no exception handling here? I would expect it here more than above in `request_password_reset`

Reply:
Added missing exception handling here.